### PR TITLE
feat(GraphCache): cache xpu_graph optimized fx graph

### PR DESCRIPTION
**Pull request by yfchen-byted**
* Add: cache xpu_graph optimized fx graph after common passes
* Perf: reduce warmup time by ~82s for model1 and ~22s for model2; enbale artifact distribution for xpu_graph backends

### CODE REVIEW

File: xpu_graph/compiler.py [hunk 1]
```diff
@@ -39,12 +51,36 @@ def _compiler(gm, sample_inputs):
                 for x in sample_inputs
             ]
             fake_mode.allow_non_fake_inputs = True
+
+            if self._cache_path is not None:
+                hashkey = hashlib.md5(
```
Comment (zhangjihang-BD) [inline xpu_graph/compiler.py:56]: 话说这个cache要不额外怎加一个模块或着class来实现吧，现在compiler有点太面向过程了

File: xpu_graph/compiler.py [hunk 2]
```diff
@@ -11,9 +11,17 @@
...
+from torch.utils._python_dispatch import _disable_current_modes
+import os
+import hashlib
+import pickle
+
 
 class XpuGraph:
-    def __init__(self, config: XpuGraphConfig = XpuGraphConfig()):
+    def __init__(
+        self, config: XpuGraphConfig = XpuGraphConfig(), cache_path: os.PathLike = None
```
[the remark this answers is not in the document]
Comment (yfchen-byted) [reply xpu_graph/compiler.py]: 已更新，默认使用default_cache作为cache，只有使用基类XpuGraphCache时才不cache

File: xpu_graph/compiler.py (hunk 1 above)
Comment (yfchen-byted) [reply xpu_graph/compiler.py]: 已添加XpuGraphCache类

**Review by zhangjihang-BD — APPROVED** [review]
LGTM
